### ISSUE TITLE
[BUG] exclude doesn't work

### ISSUE THREAD

**Issue by PiN73** [dart-code-checker/dart-code-metrics#279]
* **Dart code metrics version:** 3.0.0
* **Dart sdk version:** 2.12.3 (stable)

<details>
<summary>Configuration</summary>

```yaml
analyzer:
  exclude:
    - ios/**
    - lib/**.freezed.dart
  plugins:
    - dart_code_metrics

dart_code_metrics:
  rules:
    - prefer-trailing-comma
```

</details>

**Source code example causing the issue:** lib/hello.freezed.dart

```dart
final _privateConstructorUsedError = UnsupportedError(
    'It seems like you constructed your class using `MyClass._()`. This constructor is only meant to be used by freezed and you are not supposed to need it nor use it.\nPlease check the documentation here for more informations: https://github.com/rrousselGit/freezed#custom-getters-and-methods');
```

This code is copied from [freezed](https://pub.dev/packages/freezed) generated file, which is used in my actual project.

**What did you expect to happen?**
Problems shouldn't be reported for lib/hello.freezed.dart file as I excluded lib/**.freezed.dart

**What actually happened?**
Dart Analyzer tab reports problem in this file:

![](https://user-images.githubusercontent.com/18294052/115905816-4ae85f80-a46f-11eb-974d-886a7b632082.png)

**Note:** Default Dart problems like this
```
info: The declaration '_privateConstructorUsedError' isn't referenced. (unused_element at [flutter_app] lib\hello.freezed.dart:1)
```
successfully disappears when I use `exclude`.

**Are you willing to submit a pull request to fix this bug?**
Not sure as I don't know where is the bug comes from

**Comment by furaiev** [comment]
try to wrap your exclude params into quotation marks:

`- "**/*.freezed.dart"`

**Comment by PiN73** [comment]
Wrapping with `"` or `'` doesn't help.

Also this is only reproduced on Windows. On macOS `exclude` works. So I think it's duplicate of #268

**Comment by incendial** [comment]
@PiN73 Hi, yeah, it looks like a duplicate.

**Comment by PiN73** [comment]
Fixed in #298

**Comment by jangruenwaldt** [comment]
Still doesn't work - my config on mac, plugin 2.4.1, dart 2.14.0

```
  rules-exclude:
    - integration_test/** 
```

and reports tons of errors in integration_test, quite annoying.

**Comment by dkrutskikh** [comment]
@jangruenwaldt rules-exclude landed in 4.3.0-dev.1 version, please try to use this version

**Comment by jangruenwaldt** [comment]
> @jangruenwaldt rules-exclude landed in 4.3.0-dev.1 version, please try to use this version

Thanks for the fast answer, will give it a try and report

**Comment by jangruenwaldt** [comment]
Is this supposed to work? Because for lib/models/user/user.dart _long-parameter-list_ still is marked as error

```
  anti-patterns:
    - long-method
    - long-parameter-list:
        exclude:
          - lib/models/**
```

**Comment by dkrutskikh** [comment]
@jangruenwaldt currently anti-patterns not support exclude per pattern, feel free to open proposal

**Comment by jangruenwaldt** [comment]
> @jangruenwaldt currently anti-patterns not support exclude per pattern, feel free to open proposal

Ah too bad, that would be great to have indeed.
